Documentation/ceph-object-store-crd.md

```diff
@@ -92,16 +92,23 @@ When the `zone` section is set pools with the object stores name will not be cre
 The gateway settings correspond to the RGW daemon settings.
 
 * `type`: `S3` is supported
-* `sslCertificateRef`: If the certificate is not specified, SSL will not be configured. If specified, this is the name of the Kubernetes secret that contains the SSL certificate to be used for secure connections to the object store. Rook will look in the secret provided at the `cert` key name. The value of the `cert` key must be in the format expected by the [RGW service](https://docs.ceph.com/docs/master/install/ceph-deploy/install-ceph-gateway/#using-ssl-with-civetweb): "The server key, server certificate, and any other CA or intermediate certificates be supplied in one file. Each of these items must be in pem form."
+* `sslCertificateRef`: If specified, this is the name of the Kubernetes secret that contains the SSL certificate to be used for secure connections to the object store. Rook will look in the secret provided at the `cert` key name. The value of the `cert` key must be in the format expected by the [RGW service](https://docs.ceph.com/docs/master/install/ceph-deploy/install-ceph-gateway/#using-ssl-with-civetweb): "The server key, server certificate, and any other CA or intermediate certificates be supplied in one file. Each of these items must be in pem form."
 * `port`: The port on which the Object service will be reachable. If host networking is enabled, the RGW daemons will also listen on that port. If running on SDN, the RGW daemon listening port will be 8080 internally.
-* `securePort`: The secure port on which RGW pods will be listening. An SSL certificate must be specified.
+* `securePort`: The secure port on which RGW pods will be listening. An SSL certificate must be specified either via `sslCerticateRef` or `serviceAnnotations`
 * `instances`: The number of pods that will be started to load balance this object store.
 * `externalRgwEndpoints`: A list of IP addresses to connect to external existing Rados Gateways (works with external mode). This setting will be ignored if the `CephCluster` does not have `external` spec enabled. Refer to the [external cluster section](ceph-cluster-crd.md#external-cluster) for more details.
 * `annotations`: Key value pair list of annotations to add.
 * `labels`: Key value pair list of labels to add.
 * `placement`: The Kubernetes placement settings to determine where the RGW pods should be started in the cluster.
 * `resources`: Set resource requests/limits for the Gateway Pod(s), see [Resource Requirements/Limits](ceph-cluster-crd.md#resource-requirementslimits).
 * `priorityClassName`: Set priority class name for the Gateway Pod(s)
+* `serviceAnnotations`: Key value pair list of annotations to add for the RGW service. For example TLS secrets can be passsed to RGW with following:
+```yaml
+gateway:
+  serviceAnnotations:
+    #the feature mentioned below is supported in openshift known as service serving cert
+    service.beta.openshift.io/serving-cert-secret-name: <name of tls secret yet to generate>
```

Comment:
"name of tls secret **yet to generate**" doesn't make grammatical sense to me. Can this be reworded to be more clear?

Additionally, I think it will be important for users to understand 2 things:
1. is "service" here referring to a kubernetes Service?
2. is "secret" here referrring to a kubernetes Secret?

Reply:
The feature name is `service serving cert`, I will make it as link reference